File: WorkSessons/WorkSesson8Lesson1.lua
Options = {
    "Rock",
    "Paper",
    "Sissors",
}


local UserPlaying = true

function MainGame()
    
    print("Great, welcome to the game now type in either Rock Paper or Sissors")
    UserChoice = io.read()
    local BotChoice = math.random(1, #Options)
    for k, v in pairs(Options) do
        if UserChoice == v then
            print("The Bot Chooses "..Options[BotChoice])
            

            if BotChoice == 1 then -- if bot chooses rock
                if UserChoice == "Rock" then
                    print("You both choose rock you tied")
                elseif UserChoice == "Paper" then
                    print("You won you choose paper")
                elseif UserChoice == "Sissors" then
                    print("You lost you choose sissors")
                end
            end
            if BotChoice == 2 then -- if bot chooses paper
                if UserChoice == "Rock" then
                    print("You lost you choose rock")
                elseif UserChoice == "Paper" then
                    print("You both choose paper you tied")
                elseif UserChoice == "Sissors" then
                    print("You won you choose sisssors")
                end
            end
            if BotChoice == 3 then -- if bot chooses sissors
            if UserChoice == "Rock" then
                    print("You won you choose rock")
                elseif UserChoice == "Paper" then
                    print("You won you choose paper")
                elseif UserChoice == "Sissors" then
                    print("you both choose sissors you tied")
                end
            end

            print("Returning to startingMenu")
            StartingMenu()
            break
            return
        end
    end
    if UserPlaying == true then
        print("nope you need to type Rock Paper or Sissors try again")
        MainGame()
    end
end



function StartingMenu()
    print("Welcome to the Rock paper sissors game would you like to start? type Yes if you do otherwise type Quit if you want to exit")
    UserChoice = io.read()

    if UserChoice == "Yes" then
        print("Taking to main game")
        MainGame()
    elseif UserChoice == "Quit" then
        print("Goodbye, have a good day!")
        UserPlaying = false
        return
    elseif UserChoice ~= "Yes" and UserChoice ~= "Quit" then
        print("Try again you need to type either Yes or Quit")
        StartingMenu()
    end
end


if UserPlaying == true then
    StartingMenu()
end

File: WorkSessons/WorkSesson8Lesson1.py
import random

OptionsStorage = [
    "Rock",
    "Paper",
    "Sissors",
]


def MainGame():
    print("Welcome to the MainGame function! please type a option of Rock Paper or Sissors")

    UserChoice = input()

    RandomBotChoice = random.choice(OptionsStorage)
    

    for Option in OptionsStorage:
        if UserChoice == Option:

            if UserChoice == RandomBotChoice:
                print(f"You both picked {RandomBotChoice} you tied!")
            elif RandomBotChoice == "Rock":
                if UserChoice == "Paper":
                    print(f"You won bot choose {RandomBotChoice} and you choose {UserChoice}")
                elif UserChoice == "Sissors":
                    print(f"You lost bot choose {RandomBotChoice} and you choose {UserChoice}")

            elif RandomBotChoice == "Paper":
                if UserChoice == "Sissors":
                    print(f"You won bot choose {RandomBotChoice} and you choose {UserChoice}")
                elif UserChoice == "Rock":
                    print(f"You lost bot choose {RandomBotChoice} and you choose {UserChoice}")

            elif RandomBotChoice == "Sissors":
                if UserChoice == "Rock":
                    print(f"You won bot choose {RandomBotChoice} and you choose {UserChoice}")
                elif UserChoice == "Paper":
                    print(f"You lost bot choose {RandomBotChoice} and you choose {UserChoice}")

            print("returning to StartGame")
            StartGame()
            break
        elif UserChoice == Option:
            print("nope doesn't work try again")
            MainGame()
            

def StartGame():
    print("Welcome to rock, paper, sissors would you like to play if you would type Yes if not then type Quit")
    UserChoice = input()

    if UserChoice == "Yes":
        MainGame()
    elif UserChoice == "Quit":
        print("No worries have a good rest of your day")
    else:
        print("You need to type either Yes or Quit try again")
        StartGame()



StartGame()